### PR TITLE
add cidrs to noproxy

### PR DESCRIPTION
**Pull request by zhiweiyin318**
issue: https://github.com/open-cluster-management/backlog/issues/17702
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/controller/klusterletaddon/globalproxy_controller.go [hunk 1]
```diff
@@ -241,10 +296,49 @@ func getGlobalProxyInInstallConfig(installConfig []byte) (agentv1.ProxyConfig, e
...
+	if proxyConfig.HTTPProxy == "" && proxyConfig.HTTPSProxy == "" && noProxy == "" {
+		return proxyConfig, nil
+	}
+
+	if noProxy == "*" {
+		proxyConfig.NoProxy = noProxy
+		return proxyConfig, nil
+	}
+
+	//noProxyList := []string{".cluster.local", ".svc", "localhost", "127.0.0.1"}
```
Comment (qiujian16) [inline pkg/controller/klusterletaddon/globalproxy_controller.go:314]: add a doc here on why we need to add pod ip range etc here.

File: pkg/controller/klusterletaddon/globalproxy_controller.go [hunk 2]
```diff
@@ -241,10 +296,49 @@ func getGlobalProxyInInstallConfig(installConfig []byte) (agentv1.ProxyConfig, e
...
+	noProxyList := sets.NewString(".cluster.local", ".svc", "localhost", "127.0.0.1")
+	noProxyList.Insert(noProxy)
+
+	clusterNetworkCIDRs, err := getClusterNetworkCIDRs(proxyConfigRaw)
+	if err != nil {
```
Comment (qiujian16) [inline pkg/controller/klusterletaddon/globalproxy_controller.go:319]: what if clusterNetworkCIDRs is not found or user does not set here?

File: pkg/controller/klusterletaddon/globalproxy_controller.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/controller/klusterletaddon/globalproxy_controller.go]: done

File: pkg/controller/klusterletaddon/globalproxy_controller.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/controller/klusterletaddon/globalproxy_controller.go:340]: clusterNetworkCIDRs is required when create the cluster. 
there will meet error if clusterNetworkCIDRs is not found. will not append the proxy config to the status of klusterletaddonconfig, and output error info in the status condition.

**Pull request merged**